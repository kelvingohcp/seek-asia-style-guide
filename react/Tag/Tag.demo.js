import Tag from './Tag';
import * as sketch from './Tag.sketch';
export default {
  route: '/tag',
  title: 'Tag',
  component: Tag,
  sketch,
  initialProps: {
    label: 'This is a Tag'
  },
  options: []
};
