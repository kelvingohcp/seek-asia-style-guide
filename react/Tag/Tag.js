import styles from './Tag.less';
import React from 'react';
import PropTypes from 'prop-types';
import Text from '../Text/Text';

export default function Tag({
  label
}) {
  return (
    <span className={styles.root}>
      <Text className={styles.tagLabel} whistling strong>{label}</Text>
    </span >
  );
}

Tag.propTypes = {
  label: PropTypes.string.isRequired
};

