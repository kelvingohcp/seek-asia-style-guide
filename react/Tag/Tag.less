@import (reference) "~seek-asia-style-guide/theme";

@tag-h-padding: @grid-base * 2;
@tag-v-padding: @grid-base;

.borderRadius(@radius) {
  -webkit-border-radius: @radius;
  -moz-border-radius: @radius;
  -o-border-radius: @radius;
  border-radius: @radius;
}

.root {
  box-sizing: border-box;
  .borderRadius(@grid-base);
  color: @saGrey1;
  background-color: screen(@actionBlue, @saGrey4);
  border-color: screen(@actionBlue, @saGrey4);
  display: inline-block;
  height: @grid-base * 6;
  padding: @tag-v-padding @tag-h-padding;
  white-space: nowrap;
}

.tagLabel {
  padding-bottom: 0;
  -webkit-transform: none !important;
  transform: none !important;
}
