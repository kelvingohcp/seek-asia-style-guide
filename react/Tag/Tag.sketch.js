import React from 'react';
import mapKeys from 'lodash/mapKeys';
import Tag from './Tag';
export const listItem = {
  'Default': <Tag label='This is a Tag' />
};

// Export text styles as symbols
// (at least until Sketch libraries support shared text styles)
export const symbols = mapKeys(listItem, (value, key) => `Tag/${key}`);
