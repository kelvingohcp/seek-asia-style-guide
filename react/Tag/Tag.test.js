import React from 'react';
import { shallow } from 'enzyme';
import Tag from './Tag';

describe('Tag:', () => {
  it('should render Tag with label "This is a Tag"', () => {
    const wrapper = shallow(<Tag label='This is a Tag' />);
    expect(wrapper).toMatchSnapshot();
  });
});

