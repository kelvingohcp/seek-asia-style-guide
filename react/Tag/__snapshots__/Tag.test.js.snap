// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag: should render Tag with label "This is a Tag" 1`] = `
<span
  className="root"
>
  <Text
    baseline={true}
    className="tagLabel"
    strong={true}
    whistling={true}
  >
    This is a Tag
  </Text>
</span>
`;
